### PR TITLE
 Better naming for CacheMetadata identity function parameters.

### PR DESCRIPTION
**Pull request by coollog**
Fixes https://github.com/GoogleContainerTools/jib/pull/508#pullrequestreview-135980314

### CODE REVIEW

**Comment by coollog** [comment]
@briandealwis Looks nice! I'll update.

**Comment by coollog** [comment]
Hmm, how does this look?

File: jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheMetadata.java [hunk 1]
```diff
@@ -75,28 +78,31 @@ static boolean doLayerEntriesMatchMetadataEntries(
...
-                    .anyMatch(isNotEqual -> isNotEqual);
-              })
-          .noneMatch(isNotEqual -> isNotEqual);
+      return !pairwiseCompareAnyTrue(
+          layerEntries,
+          metadataEntries,
+          (layerEntry, metadataEntry) -> {
+            // Checks extraction path not equal.
+            if (!layerEntry.getExtractionPath().equals(metadataEntry.getExtractionPath())) {
+              return true;
```
Comment (briandealwis) [inline jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheMetadata.java]: `return true` on not-equals is counter-intuitive.

File: jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheMetadata.java [hunk 2]
```diff
@@ -75,28 +78,31 @@ static boolean doLayerEntriesMatchMetadataEntries(
...
+            }
+
+            // Checks for any source file not equal.
+            if (layerEntry.getSourceFiles().size()
+                != metadataEntry.getSourceFilesStrings().size()) {
+              return true;
```
Comment (briandealwis) [inline jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheMetadata.java]: `return true` on not-equals are counter-intuitive.

File: jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheMetadata.java [hunk 3]
```diff
@@ -75,28 +78,31 @@ static boolean doLayerEntriesMatchMetadataEntries(
...
+            }
+            return pairwiseCompareAnyTrue(
+                layerEntry.getSourceFiles(),
+                metadataEntry.getSourceFilesStrings(),
+                (sourceFile, sourceFileString) -> !sourceFile.equals(Paths.get(sourceFileString)));
```
Comment (briandealwis) [inline jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheMetadata.java]: This is weird.

**Comment by coollog** [comment]
I changed it all to your original suggestion. I don't know what I was thinking because that works fine haha.

**Pull request merged**